File: 24_C#Fundamentals/_05_Lists/_01_Lecture/_02_GaussTrick/_02_GaussTrick.cs
using System;
using System.Xml.Linq;
using System.Globalization;

public class _02_GaussTrick
{
    public static void Main(string[] args)
    {
        // Read input values
        List<int> listNumbers = Console.ReadLine().Split().Select(int.Parse).ToList();

        // Add variable
        int middle = listNumbers.Count / 2;

        // Sum numbers in the given pattern
        for (int i = 0; i < middle; i++)
        {
            listNumbers[i] = listNumbers[i] + listNumbers[listNumbers.Count - 1];
            listNumbers.RemoveAt(listNumbers.Count - 1);
        }

        //System.out.println(listNumbers);
        Console.WriteLine(string.Join(" ", listNumbers));
    }
}

File: 02_ProgrammingFundamentals/_05_Lists/_01_Lecture/_02_GaussTrick.java
package _02_ProgrammingFundamentals._05_Lists._01_Lecture;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class _02_GaussTrick {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        List<Integer> listNumbers = Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());
        // List<String> hmm = Arrays.stream(scan.nextLine().split("\\s+")).collect(Collectors.toList());

        // Add variable
        int middle = listNumbers.size() / 2;

        // Sum numbers in the given pattern
        for (int i = 0; i < middle; i++) {
            listNumbers.set(i, listNumbers.get(i) + listNumbers.get(listNumbers.size() - 1));
            listNumbers.remove(listNumbers.size() - 1);
        }

        //System.out.println(listNumbers);
        listNumbers.stream().forEach(e -> System.out.print(e + " "));

        scan.close();
    }
}
